### ISSUE TITLE
expiryMonth truncated in card data

### ISSUE THREAD

**Issue by pszalko** [tpay-com/omnipay-tpay#7]
Card expire date should be in format `MM/YYYY` according to official documentation regarding card cipher:

> Card cipher is made from string:
> card number|expiry date(MM/YY or MM/YYYY)|cvv or cvc|host eg. "1234567891234567|05/17|123|https://merchantwebsite.com"

However, this library does not work as expected and produce incorrect card cipher value for dates with expiry month in range `01-09`.

I have found that Omnipay code (which is baseline for this lib) casts `Month` value to `INT` in method:
`\Omnipay\Common\CreditCard::setExpiryMonth()`:

```PHP
    public function setExpiryMonth($value)
    {
        return $this->setParameter('expiryMonth', (int) $value);
    }
```

Please fix this issue as it produce unexpected results for transactions with cards with expiry month in range `01` - `09`.

**Comment by piotrjozwiak** [comment]
@pszalko Omnipay methods casts expiry month and year to integer, so after setting example card data we get:

`number' => string '4111111111111111' (length=16)
'expiryMonth' => int 8
'expiryYear' => int 2019
'cvv' => int 502`

So it is true, that Tpay.com library calculates card hash from string eg.

> 4111111111111111|8/2019|502|http://localhost

But this situation is handled on our server side, so this should not cast any problems with processing payments.
Did you get any errors?

**Comment by pszalko** [comment]
Yes, we have errors when processing payments.

None of the real payment cards we tested on production environment with 3D secure enabled and expiry month in range `01 - 09` works. 

There are few issues we found:
- sometimes transaction is refused with "3D error" visible only on TPay transactions list (user couldn't enter this code on Bank website)
- sometimes transaction is not even visible in TPay transactions list
- sometimes Bank website shows immediate error message instead of 3D Secure web form
- sometimes Bank form to enter 3D Secure code is not shown at all and TPay returns error for 3D secure card before redirecting to Bank website

However, if we try payment with card expiring in month `10` transaction is successful in the same Bank as tested cards.

We tried 3 different polish Banks and cards with 3D secure enabled.

**Comment by piotrjozwiak** [comment]
Ok thank you for the detailed feedback, this may be very useful for us.
I have committed quick fix for this issue 9dc146139f1e2ab56f953b1a94c5343f37a769dc

**Comment by pszalko** [comment]
Thanks, this quick fix resolves the issue for us.

However, as you can see the error was propagated to Bank website, which I assume, is deeper problem. I have also contacted Aleksandra from TPay support earlier, so she will have more information about it.

If you need more details regarding our investigation to fix other possible issues, contact me at: przemek@rentcars.pl
